packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -663,39 +661,42 @@ export function requestOps(
  * @internal
  */
 export const emptyMessageStream: IStream<ISequencedDocumentMessage[]> = {
-	read: async () => {
+	read: async (): Promise<IStreamResult<ISequencedDocumentMessage[]>> => {
 		return { done: true };
 	},
 };
 
 /**
+ * Creates a stream from a promise of messages.
  * @internal
  */
 export function streamFromMessages(
 	messagesArg: Promise<ISequencedDocumentMessage[]>,
 ): IStream<ISequencedDocumentMessage[]> {
 	let messages: Promise<ISequencedDocumentMessage[]> | undefined = messagesArg;
 	return {
-		read: async () => {
+		read: async (): Promise<IStreamResult<ISequencedDocumentMessage[]>> => {
 			if (messages === undefined) {
 				return { done: true };
 			}
-			const value = await messages;
+			const currentMessages = messages;
```

Comment:
Add a small comment about why do it this way? Something related to atomicity, IIRC?